### DISCUSSION TITLE
Old citation key format migration to newer version

### DISCUSSION THREAD

**Discussion by John-foobar-Doe** [retorquere/zotero-better-bibtex#2135] [Q&A] [answered]
I have the following citation key format in Zotero preferences UI (with Better BibTeX 1.6.100 installed alongside Zotero 4 for Firefox):
![image](https://user-images.githubusercontent.com/104580395/165785882-10314eaf-84f5-4ad6-a405-5c922b64b1b7.png)

Now I'm migrating to Zotero 6 with Better BibTeX 6.5.2 installed, where the default citation key format is:
![image](https://user-images.githubusercontent.com/104580395/165786101-38b1be85-7e11-42bd-8209-d1f2079ea88e.png)

So now, can I simply copy & paste the old key format `[auth.etal]_[year]_[shorttitle]` to Zotero 6 preferences UI window's text field? Will Better BibTeX 6.5.2 still recognize older citation key format? If not, what key should I use to replicate the function of the old key `[auth.etal]_[year]_[shorttitle]`? Many thanks.

Other request: for backward compatibility issue like this, could you please add relevant information to pages such as https://retorque.re/zotero-better-bibtex/citing/, such that users won't need to ask here and wait for replies/answers (sometimes users like me need to resolve this as soon as possible and can't afford waiting time)? Many thanks again.

**Comment by retorquere** [comment, marked as answer]
I'll amend the docs like your say, but you can just paste in your existing pattern. BBT will show what the new format is, but the old format works as-is. Should it not, that would be a bug.

**Reply by retorquere** [reply]
The equivalent to the pattern you have is

```
authEtal2 + '_' + year + '_' + shorttitle
```

although I'd say

```
[auth.etal:postfix=_][year:postfix=_][shorttitle] 
```

is a better formula, and the equivalent new format would be

```
authEtal2.postfix(_) + year.postfix(_) + shorttitle
```

**Reply by John-foobar-Doe** [reply]
Thanks a lot!

**Comment by aemartinez** [comment]
Hi, I have a similar issue. My previous citation key format was: 

`[=journalArticle][auth:lower]:[PublicationTitle:skipwords:abbr:lower:alphanum][volume:prefix=-][issue:prefix=_] | [=conferencePaper][auth:lower]:[extra=tex.shortconf:lower:>0][shortyear] | [=conferencePaper][auth:lower]:[conferenceName:skipwords:abbr:lower:alphanum][shortyear] | [auth:lower][shortyear]`

Which worked flawlessly before I updated Zotero and Better Bibtex. Now it yields the following error:

`Line 1: Unexpected token =`

Should I submit this as an issue? 

Thanks for maintaining this wonderful tool!
Best,
Agustín.

**Reply by retorquere** [reply]
Yes please.
